lib/ngx/ssl.md

```diff
@@ -299,6 +300,56 @@ This function can be called in whatever contexts where downstream https is used.
 
 [Back to TOC](#table-of-contents)
 
+raw_client_addr
+---------------
+**syntax:** *addr, err = ssl.raw_client_addr()*
+
+**context:** *any*
+
+Returns the raw client address of the current SSL connection.
+
+The first two return values are strings representing the address data and the address type, respectively.
+The address values are interpreted differently according to the address type values:
+
+* `unix`
+: The address data is a file path for the UNIX domain socket.
+* `inet`
+: The address data is a binary IPv4 address of 4 bytes long.
+* `inet6`
+: The address data is a binary IPv6 address of 16 bytes long.
+
+Returns two `nil` values and a Lua string describing the error.
+
+The following code snippet shows how to print out the UNIX domain socket address and
+the IPv4 address as human-readable strings:
+
+```lua
+local ssl = require "ngx.ssl"
+local byte = string.byte
+
+local addr, addrtyp, err = ssl.raw_client_addr()
+if not addr then
+    ngx.log(ngx.ERR, "failed to fetch raw client addr: ", err)
+    return
+end
+
+if addrtyp == "inet" then  -- IPv4
+    ip = string.format("%d.%d.%d.%d", byte(addr, 1), byte(addr, 2),
```

Comment:
@smartwjw 
Better to localize `string.format` just like the `string.byte`?

Reply:
Agreed here.